.github/workflows/elixir.yml

```diff
@@ -18,15 +18,52 @@ permissions:
   contents: read
 
 jobs:
+  formatting-check:
+    runs-on: ubuntu-latest
+    name: Formatting check
+    steps:
+      - name: Set up Elixir
+        uses: erlef/setup-beam@v1
+        with:
+          otp-version: 25.3.2
+          elixir-version: 1.14.4
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Cache deps
+        id: cache-deps
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-elixir-deps
+        with:
+          path: |
+            deps
+            _build
+
+          key: ${{ runner.os }}-mix-${{ env.cache-name }}-${{ hashFiles('**/mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-mix-${{ env.cache-name }}-
+
+      - name: Install dependencies
+        run: |
+          mix deps.get
+
+      # Step: Check that the checked in code has already been formatted.
+      # This step fails if something was found unformatted.
+      # Customize this step as desired.
+      - name: Check Formatting
+        run: mix format --check-formatted
```

Comment:
Why this change? We have a precommit hook and testing task that should prevent unformatted code from being merged into main. 

Also, I don't think you need to fetch and compile deps just to check formatting

Reply:
@scohen I found that format style diff on Elixir version 1.13 T_T (https://github.com/lexical-lsp/lexical/actions/runs/4938201783/jobs/8827646922). So to avoid this I extract to a new job.

Reply:
Or add some boolean in matrix (like `check-format: true | false`) and `if` at this step would be better?

Reply:
how about we just format it so that 1.13 doesn't complain?